### PR TITLE
Attempting to clean up some remaining Solr references

### LINKED ISSUE #12934: Should we clean up the few remaining references to `Lucene/Solr`?
### Description

I was looking at a very old issue for a failing unit test with an ant test command. When I tried to run it in gradle (incorrectly), I was greeted with a message that stood out to me:
```
This is an experimental Lucene/Solr gradle build.
```

Searching the Lucene codebase ([github search](https://github.com/search?q=repo%3Aapache%2Flucene%20Lucene%2FSolr&type=code)), we find 18 usages of `Lucene/Solr`, all in comments/readmes/print messages in tools, most of these can be switched to just `Lucene`

Should we clean this up? If yes, I am happy to do it :)

### PR DESCRIPTION
**Pull request by slow-J**
Cleaned up some old references to Solr in the codebase, including references to some Solr packages and directories in tests.
Please let me know if these changes would affect anything in Solr itself?

Closes https://github.com/apache/lucene/issues/12934.

### CODE REVIEW

File: gradle/help.gradle [hunk 1]
```diff
@@ -46,7 +46,7 @@ configure(rootProject) {
   help {
     doLast {
       println ""
-      println "This is an experimental Lucene/Solr gradle build. See some"
+      println "This is an experimental Lucene gradle build. See some"
```
Comment (dweiss) [inline gradle/help.gradle:49]:
```suggestion
      println "This is Lucene's gradle build. See some"
```

**Review by dweiss — APPROVED** [review]
Looks good to me. If anything removed is used by Solr then it's perhaps the highest time to tweak it over downstream.

**Comment by slow-J** [comment]
> Looks good to me. If anything removed is used by Solr then it's perhaps the highest time to tweak it over downstream.

Thanks for the review @dweiss!

**Review by mikemccand — APPROVED** [review]
Thanks @slow-J for the attention to details -- looks great -- I'll merge soon.

**Pull request merged**